File: ui/tract-renderer/src/main/java/org/cru/godtools/tract/viewmodel/ImageViewHolder.java
package org.cru.godtools.tract.viewmodel;

import android.view.ViewGroup;

import org.ccci.gto.android.common.picasso.view.PicassoImageView;
import org.cru.godtools.base.tool.model.view.ResourceViewUtils;
import org.cru.godtools.tract.R;
import org.cru.godtools.tract.R2;
import org.cru.godtools.xml.model.Image;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.annotation.UiThread;
import butterknife.BindView;
import butterknife.OnClick;

@UiThread
final class ImageViewHolder extends BaseViewHolder<Image> {
    @BindView(R2.id.image)
    PicassoImageView mImage;

    ImageViewHolder(@NonNull final ViewGroup parent, @Nullable final BaseViewHolder parentViewHolder) {
        super(Image.class, parent, R.layout.tract_content_image, parentViewHolder);
    }

    @Override
    void onBind() {
        super.onBind();
        ResourceViewUtils.bind(Image.getResource(mModel), mImage);
    }

    @OnClick(R2.id.image)
    void click() {
        if (mModel != null) {
            sendEvents(mModel.getEvents());
        }
    }
}

File: ui/tract-renderer/src/main/java/org/cru/godtools/tract/viewmodel/ImageViewHolder.kt
package org.cru.godtools.tract.viewmodel

import android.view.ViewGroup
import androidx.annotation.UiThread
import butterknife.BindView
import butterknife.OnClick
import org.ccci.gto.android.common.picasso.view.PicassoImageView
import org.cru.godtools.base.tool.model.view.ResourceViewUtils
import org.cru.godtools.tract.R
import org.cru.godtools.tract.R2
import org.cru.godtools.xml.model.Image
import org.cru.godtools.xml.model.resource

@UiThread
internal class ImageViewHolder(parent: ViewGroup, parentViewHolder: BaseViewHolder<*>?) :
    BaseViewHolder<Image>(Image::class.java, parent, R.layout.tract_content_image, parentViewHolder) {
    @BindView(R2.id.image)
    lateinit var image: PicassoImageView

    public override fun onBind() {
        super.onBind()
        ResourceViewUtils.bind(mModel.resource, image)
    }

    @OnClick(R2.id.image)
    fun click() {
        mModel?.events?.let { sendEvents(it) }
    }
}
